client/components/board/simulatorFactory.js

```diff
@@ -0,0 +1,32 @@
+angular.module('OpenEXP')
+    .factory('simulatorFactory', () => {
+        const OpenBCIBoard = require('openbci-sdk');
+        const ourBoard = new OpenBCIBoard.OpenBCIBoard();
+
+        var storage = {};
+
+        var simulatorStart = () => {
+            ourBoard.simulatorStart()
+                .then(() => {
+                    ourBoard.on('sample',sample => {
+                        storage[sample.sampleNumber] = sample;
+                    })
+                }).catch(err => console.log('Error [simulator]: ' + err))
+        };
+
+        var observer = function(changes) {
```

Comment:
we turn on the simulator with simulatorStart(), then once we receive a 'sample', which means the simulator is started, we add to the storage object the value of sample. So inside storage are a bunch of keys that are the sampleNumbers, 1,2,3,4... whose value is the entire sample. Just a easy way of storing. The observer is essentially a watcher that just dirty checks storage to see if anything has changed, if it has then it fires